src/rely/TestSuiteRunner.re

```diff
@@ -0,0 +1,297 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */;
+open TestSuite;
+open TestResult;
+open CommonCollections;
+open MatcherUtils;
+open Test;
+open Describe;
+open Util;
+open RunConfig;
+open Reporter;
+exception PendingTestException(string);
+module SnapshotModuleSet =
+  Set.Make({
+    type t = (module Snapshot.Sig);
+    let compare = (a, b) => a === b ? 0 : (-1);
```

Comment:
I think this will cause problems because it's not a true ordering. (At least you will not get any perf benefits from Set). If It's always returning -1 or 0 then nodes can only be added and searched for as a left child in the tree and you will build a long linked list and get O(n) lookup times.

Can you do a true less than/greater than compare function here?